### DISCUSSION TITLE
Fail to import skin textures from Blender to CC3 correctly

### DISCUSSION THREAD

**Discussion by Pizhlo** [soupday/cc3_blender_tools#72] [Q&A]
I made some character in CC3. Then I imported this character in Blender using the CC3 add-on for Blender. Now I finished my work in Blender and need to import my character back to CC3. So I faced two problems:

1. When importing my character from Blender to CC3 (using Create > Cloth, Hair, Acc.. button) textures of skin, eyes, teeth, tongue don't appear. I found out that I need to make a new key using CC3 add-on for Blender (is it true?). And here is the 2nd issue.
2. The button 'export to CC3' in Blender menu is not available. 

How do I solve these problems? I guess if I solve the second issue I solve the first one also. But how to do it?

PS I've just downloaded the latest verison of CC3 add-on for Blender but the issue still isn't solved.

**Comment by soupday** [comment]
1. If CC3 let you import your character back in with the Create > Cloth... menu then you have a valid FbxKey. The add-on does not and cannot make an FbxKey, only CC3 can do that with the initial export.

2. Since you have a valid key, did you import the character with the 'Import Character' button in the add-on panel?

**Reply by Pizhlo** [reply]
2. Add-on for CC3 doesn't work for some reasons (I've made a discussion about this problem also, see discussion#73) so I can't use it.

**Reply by soupday** [reply]
2. The add-on in Blender. If you don't use the button in the add-on panel to import the character into Blender, then it's not using the add-on. It can't pick up generic FBX imports, from the import menu, because it's not possible to tell where they came from.

**Reply by Pizhlo** [reply]
I use addon for Blender to import my character from CC3 to Blender. Now when i need to export my character from Blender to CC3, I can't use the add-on to do it because the export button isn't available. Why does it happen?

**Reply by soupday** [reply]
Is there an .Fbxkey file with the initial export from CC3?

You should have:

character_name.fbx
character_name.json
character_name.fbxkey
character_name.fbm/ (main textures folder)
textures/character_name/ (additional textures folder)

The add-on needs all of these to setup the character properly and export it back to CC3.

**Reply by soupday** [reply]
though only a missing fbxkey will cause the "Export To CC3" button to be greyed out.

**Reply by Pizhlo** [reply]
By the way, should the textures be visible when a character is imported from Blender to CC3? Or I have to manually overlay the textures? 

Yes, I have a Fbxkey which helps me to open my character in CC3 (the version before editing in Blender). It works correctly in this case. It doesn't work in the version of my character after editing in Blender.

**Reply by soupday** [reply]
By default the import back to CC3 doesn't bring all the textures with it (at least not FBX exports from Blender). The FBX format can't pick up all the textures, so it's really just for the mesh & armature data. 

It can pick up some textures (but only the diffuse, alpha & normal) if the materials are configured in a way that the FBX exporter expects, but the materials as set up by the add-on are too complex so it doesn't get any.

That's why I had to make the plugin for CC3, to rebuild the textures.

**Comment by harbuzenger** [comment]
In response to the textures not showing up in CC3 - I had renamed my files such as:

Export from CC3:

"xxxxxx_CC3_to_blender_v001"

Then I would sculpt and make changes - making sure not to change the hierarchy of the imported fbx, or the vertex order (both will mean an export from Blender will fail).

Then exporting from Blender:

"xxxxxx_blender_to_CC3_v001"

So far fine, however the textures will still not work with the CC3 plugin because the file names have changed.  So either you overwrite the original FBX (I didn't want to) or you edit the .json file that is generated by CC3 when you export from there.  I used wordpad to replace all instances of "xxxxxx_CC3_to_blender_v001" to "xxxxxx_blender_to_CC3_v001" and changed the two folder names where the textures live (in that output folder in "textures" and the .fbm folder- assuming you choose not to embed textures on the CC3 export) to the new name as well.

Renaming the output file to "xxxxxx_CC3_to_blender_v001_old" is probably the most efficient way but I wasn't sure I'd be breaking other things, so I was keeping original files intact.

**Reply by soupday** [reply]
The file name is essentially hard coded into the .json data, which is why it's not a good idea to change any file names.

Exporting from Blender via the add-on let's you change the name and it remaps all the names and file paths so the .json data matches the FBX exactly and all the file paths to the textures are valid. But for that to work, you must import into Blender using the add-on 'Import Character' Button & _not_ the file > import menu.

**Reply by harbuzenger** [reply]
Thanks @soupday but the Blender add-on doesn't change/update the .json in my experience.  I just kept getting the error inside CC3 (using your plugin) that no json data exists for this fbx.  Duplicating the .json and editing as I outlined above does work around this issue.

**Reply by soupday** [reply]
Are you renaming the fbx export from CC3? If so it won't find the json data. It all has to match otherwise it won't work.

**Reply by harbuzenger** [reply]
no, I'm renaming my file when exporting from Blender.  Workflow as follows:

Do some stuff in CC3

export from CC3: **"xxx_from_CC3_v001"** for example

use add-on to import into Blender.  Do some stuff (basic sculpting for now while I'm trying to figure out a nice workflow).

export from Blender: **"xxx_from_blender_v001"** for example.

Into the same folder though.

Then use the plugin inside CC3 to import *"xxx_from_blender_v001"*

Error - no .json file found.

Edit .json file to match *"xxx_from_blender_v001"* and update folder structure names to match *"xxx_from_blender_v001"*

No error - CC3 import behaves correctly

**Reply by soupday** [reply]
Odd that it's not picking up the json data.

Do you always use three digits at the end of the names? I wonder if the add-on is stripping them out.

To properly remap object and material names back to CC3, I have to detect when Blender has renamed anything with duplication suffixes. .001, .002 etc.. and they get stripped out so it matches the json and the fbxkey. If it's doing that to the file name...

**Reply by harbuzenger** [reply]
Just to say I've started again and used the plugin to go from CC4 to Blender.  Used the add-on to import, and then export.  Then the CC4 plugin to reimport and it works perfectly.  I hadn't been using the plugin inside CC4 to go in and out.  I think that was the issue (and generally being a bit daft).  Apologies for the late reply as well.  It has been a busy 21 days...
